src/util/cursor-get-text.ts

```diff
@@ -6,9 +6,12 @@ import { EditableDocument } from '../cursor-doc/model';
 
 export type RangeAndText = [[number, number], string];
 
-export function currentTopLevelFunction(doc: EditableDocument): RangeAndText {
-    const defunCursor = doc.getTokenCursor(doc.selection.active);
-    const defunStart = defunCursor.rangeForDefun(doc.selection.active)[0];
+export function currentTopLevelFunction(
+    doc: EditableDocument,
+    active: number
```

Comment:
Can you make the current document selection the default, maybe?

```suggestion
    active: number = doc.selection.active
```

We do that in a lot of places. I haven't looked super closely, but maybe it will alow the tests to continue working.

Reply:
I fixed all the places that use it since they have access to the doc anyway and then just pass in the `doc.selection.active`. The only place this is a problem is the tests, so I'd rather not add a default case that's not being used.

Reply:
You can see it as a way to inject the selection from the tests.

Reply:
Ah, yeah, [that's](https://github.com/BetterThanTomorrow/calva/pull/1582/commits/4bf27621c50166024bb8be759d66772345d9bfa8) a way to do it of course. :smile:

Reply:
I still think it makes sense to have a default behaviour of the current cursor position. That has worked very well for us so far.

Reply:
Ok. I'll add it